### ISSUE TITLE
💅 False error (or at least a misleading one) - private class member never used

### ISSUE THREAD

**Issue by craftedsystems** [biomejs/biome#6933]
### Environment information

```bash
CLI:
  Version:                      2.1.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.3.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.4.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useIsNan
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    performance/noAccumulatingSpread
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noInferrableTypes
    style/noParameterAssign
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useConst
    style/useDefaultParameterLast
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExponentiationOperator
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodejsImportProtocol
    style/useNumberNamespace
    style/useSelfClosingElements
    style/useShorthandFunctionType
    style/useSingleVarDeclarator
    style/useTemplate
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConsole
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useAwait
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0
```

### Rule name

lint/style/noUnusedPrivateClassMembers

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABUAG8AYQBzAHQAUwBlAHIAdgBpAGMAZQAgAHsACgAgACAAcAByAGkAdgBhAHQAZQAgAF8AdABvAGEAcwB0AEkAZAAgAD0AIAAwADsACgAKACAAIABzAGgAbwB3ACgAbQBlAHMAcwBhAGcAZQA6ACAAcwB0AHIAaQBuAGcALAAgAHQAeQBwAGUAOgAgAHMAdAByAGkAbgBnACwAIABhAHUAdABvAEMAbABvAHMAZQA6ACAAYgBvAG8AbABlAGEAbgApADoAIAB2AG8AaQBkACAAewAKACAAIAAgACAAYwBvAG4AcwB0ACAAaQBkACAAPQAgAHQAaABpAHMALgBfAHQAbwBhAHMAdABJAGQAKwArADsACgAgACAAIAAgAHQAaABpAHMALgBhAGMAdABpAHYAZQBUAG8AYQBzAHQAcwAuAHAAdQBzAGgAKAB7ACAAaQBkACwAIABtAGUAcwBzAGEAZwBlACwAIAB0AHkAcABlACwAIABhAHUAdABvAEMAbABvAHMAZQAgAH0AKQA7AAoAIAAgAH0ACgB9AAoA&language=ts

### Expected result

Biome reports a false error.

```sh
  ⚠ This private class member is defined but never used.
  
    22 │   private _root: LitElement | ShadowRoot;
    23 │   private _toastsEl: UIToasts;
  > 24 │   private _toastId = 0;
       │           ^^^^^^^^^^
    25 │ 
    26 │   /**
  
  ℹ Unsafe fix: Remove unused declaration.
  
    22 22 │     private _root: LitElement | ShadowRoot;
    23 23 │     private _toastsEl: UIToasts;
    24    │ - ··private·_toastId·=·0;
    25 24 │   
    26 25 │     /**
```

But the private class member `_toastId` is used like in this simplified code example.

```TypeScript
export class ToastService {
  private _toastId = 0;

  show(message: string, type: string, autoClose: boolean): void {
    const id = this._toastId++;
    this.activeToasts.push({ id, message, type, autoClose });
  }
}
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by vladimir-ivanov** [comment]
this has been already fixed, but not deployed yet. Will however write a test for it, to avoid future regressions.

**Comment by vladimir-ivanov** [comment]
still an issue, was looking into nursery instead of correctness. Will address

**Comment by vladimir-ivanov** [comment]
it also violates useReadonlyClassProperties:
https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=KYDwDg9gTgLgBAYwDYEMDOa4CEUBMCiIKAtmEsHAN4BQciEAdmjFAK4IzQAUYUAlgDcUMCmgjFgMABZ8GAcwBccZv3kBKKnAC%2B1WnFAkywLhpp06CRs01iJ02XO1wAvHHto9dKJNZQGy8UkZeT0dLSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wls4CGAEwD2TeIXRRe0EdEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

**Comment by agarny** [comment]
Has this really been fixed? I tried the playground with both `v2.x` and `next` and it still reports `lint/correctness/noUnusedPrivateClassMembers: This private class member is defined but never used.`

(Yes, I am having the exact same issue, using v2.3.6.)

**Comment by agarny** [comment]
FWIW, a workaround is to make the variable `protected` rather than `private`. Not great, but at least there is no more false error.

**Comment by albertodiazdorado** [comment]
Has this been checked? I am seeing the same false positives in version `2.3.6`

**Comment by mividtim** [comment]
> Has this been checked? I am seeing the same false positives in version `2.3.6`

Yes, this bug seems to have been reintroduced. In every case in my code, it seems to be member arrow functions of the class that are being ignored for this check. Hope this helps!

**Comment by gabrielslacerdaext-hue** [comment]
I'm facing it on 2.3.6 too... The issue must be open again.

**Comment by albertodiazdorado** [comment]
How do we re-open it?

**Comment by mividtim** [comment]
@vladimir-ivanov This seems to have ben reintroduced! I've had to turn the rule "off" to avoid a barrage of warnings or errors every time I run biome. We always use arrow functions for class members to avoid "this" confusion.
